### PR TITLE
Change validation data from test to training data

### PR DESCRIPTION
**Pull request by bbartoldson**
5,000/60,000 training images are set aside to use as validation data
during parameter tuning. once parameter tuning is complete, the final
model is trained on all 60,000 training images for 3 epochs. this
approach is consistent with one taken by Yann LeCun in the paper "What
is the Best Multi-Stage Architecture for Object Recognition?"

### CODE REVIEW

**Comment by fchollet** [comment]
That's would be a reasonable way to proceed, for instance in a machine learning competition where you do not have access to the score on the test set. In our case there is not much of a point. We're just using the test set as validation set --to confirm that the model is working, and how well it's working.

**Comment by bbartoldson** [comment]
Makes sense. I suggested my revision because I was confused about the absence of a separate validation set, but I understand why you didn't include it now. Thanks for explaining.

**Comment by Euphemiasama** [comment]
@fchollet @bbartoldson Is monitoring the validation loss to reduce the learning rate ,when it reaches a plateau, not considered tweaking the parameters ? So in that case aren't we peeking into the test data (validation data is the case of keras) which would eventually lead to overfitting?

**Comment by WittmannF** [comment]
I still think this example should be updated. It is bad practices to use the test set as validation set, even though it is not being used for model selection here. @fchollet, kindly check the full discussion at #1753 (especially [here](https://github.com/keras-team/keras/issues/1753#issuecomment-223784234)) and please reconsider this pull request.
